standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -1636,27 +1638,41 @@ public static String getPartitionName(Path tablePath, Path partitionPath, Set<St
     return result;
   }
 
-  public static String getNormalisedPartitionValue(String partitionValue, String type) {
-
-    if (!NumberUtils.isParsable(partitionValue)) {
+  public static String getNormalisedPartitionValue(String partitionValue, String type,
+      Configuration conf) {
+    if (!NumberUtils.isParsable(partitionValue) && !type.equalsIgnoreCase("string")
```

Comment:
don't we have constants or enum for type like string?

Reply:
Changed to used an enum